src/libraries/Common/src/Roslyn/CSharpSyntaxHelper.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
the general approach i took was to copy the roslyn code whole-sale and make as few changes to it as possible.  The only changes i made were:

1. update the license.
2. change the namespace.
3. use ValueListBuilder instead of ArrayBuilder.

Reply:
Yes. @chsienki and @jaredpar both reviewed the roslyn side. They should likely review this port as well.